tests/check/check_h26xsigned_auth.c

```diff
@@ -1625,6 +1625,310 @@ START_TEST(vendor_axis_communications_operation)
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_set_private_key(sv_camera, setting.algo, private_key, private_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  sv_rc = signed_video_add_public_key_to_sei(sv_camera, add_public_key_to_sei);
+  // Setting validation level.
+  sv_rc = signed_video_set_authenticity_level(sv_camera, setting.auth_level);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  // Add an I-NALU to trigger a SEI.
+  sv_rc = signed_video_add_nalu_for_signing(sv_camera, i_nalu->data, i_nalu->data_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_get_nalu_to_prepend(sv_camera, &nalu_to_prepend);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  *sei = nalu_list_create_item(nalu_to_prepend.nalu_data, nalu_to_prepend.nalu_data_size, setting.codec);
+  sv_rc = signed_video_get_nalu_to_prepend(sv_camera, &nalu_to_prepend);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  ck_assert(nalu_to_prepend.prepend_instruction == SIGNED_VIDEO_PREPEND_NOTHING);
+  ck_assert(tag_is_present(*sei, setting.codec, PUBLIC_KEY_TAG) == add_public_key_to_sei);
+  free(private_key);
+
+  return sv_camera;
+}
+
+static void validation_helper_function(signed_video_t *sv, nalu_list_item_t *sei, bool wrong_key, signature_info_t *signature_info, SignedVideoCodec codec)
+{
+  SignedVideoReturnCode sv_rc;
+  bool public_key_present = sv->has_public_key;
+
+  // Validate this first GOP.
+  signed_video_authenticity_t *auth_report = NULL;
+  signed_video_latest_validation_t *latest = NULL;
+
+  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
+  sv_rc = signed_video_add_nalu_and_authenticate(sv, sei->data, sei->data_size, &auth_report);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  ck_assert(!auth_report);
+
+  if (signature_info) {
+    sv_rc = signed_video_set_public_key(
+        sv, signature_info->public_key, signature_info->public_key_size);
+    ck_assert_int_eq(sv_rc, SV_NOT_SUPPORTED);
+  } else {
+
+    sv_rc = signed_video_add_nalu_and_authenticate(sv, i_nalu->data, i_nalu->data_size, &auth_report);
+
+    if (public_key_present) {
+      ck_assert_int_eq(sv_rc, SV_OK);
+    } else {
+      ck_assert_int_eq(sv_rc, SV_NOT_SUPPORTED);
+    }
+
+    if (public_key_present) {
+      ck_assert(auth_report);
+      latest = &(auth_report->latest_validation);
+      ck_assert(latest);
+
+      if (wrong_key) {
+        ck_assert(latest->public_key_has_changed);
+        ck_assert_int_eq(latest->authenticity, SV_AUTH_RESULT_NOT_OK);
+      } else {
+        ck_assert(!(latest->public_key_has_changed));
+        ck_assert_int_eq(latest->authenticity, SV_AUTH_RESULT_OK);
+      }
+    }
+    // We are done with auth_report
+    signed_video_authenticity_report_free(auth_report);
+
+    // Free nalu_list_item and session.
+    signed_video_free(sv);
+    nalu_list_free_item(sei);
+    nalu_list_free_item(i_nalu);
+  }
+}
+
+/* Test description */
+START_TEST(public_key_on_validation_side_from_start)
```

Comment:
As you probably have noticed, all tests now look very similar. You can reduce the code footprint by combining them and inside test(s) loop through the various alternatives.